flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -117,6 +120,10 @@ public void initializeState(StateInitializationContext context) throws Exception
     // Open the table loader and load the table.
     this.tableLoader.open(hadoopConf.get());
     this.table = tableLoader.loadTable();
+
+    int subTaskId = getRuntimeContext().getIndexOfThisSubtask();
+    int attemptId = getRuntimeContext().getAttemptNumber();
+    this.flinkManifestFactory = FlinkManifest.createFactory(table, flinkJobId, subTaskId, attemptId);
```

Comment:
I see the `createFactory` definition is `createFactory(Table table, String flinkJobId, int partitionId, long taskId)` . Do we need to make parameters as same as here?

Reply:
+1 on the comment. I also asked a question regarding `partitionId` in the definition place, as it is not very intuitive name. Now looking at the usage here. I am wondering if the arg order is correct. In the definition, `taskId` is the last/4th arg, while here `subtaskid` is supplied in the 3rd arg.

Reply:
It's true. we should use `subTaskId` and `attemptId` for those arguments.